src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -844,6 +846,21 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
     /*package*/ GitClient createClient(TaskListener listener, EnvVars environment, Job project, Node n, FilePath ws) throws IOException, InterruptedException {
 
         String gitExe = getGitExe(n, listener);
+
+        UnsupportedCommand unsupportedCommand = new UnsupportedCommand();
+        for (GitSCMExtension ext : extensions) {
+            ext.determineSupportForJGit(this, unsupportedCommand);
+        }
```

Comment:
The content for extensions will be the real git extension list (user chosen additional behavior) since the GitSCM constructor takes care of filling this `extensions` with the real list. 

For reference: https://github.com/jenkinsci/git-plugin/blob/4c6a328fc449121f3ddf59dcc69490208975e29c/src/main/java/hudson/plugins/git/GitSCM.java#L233

Reply:
I need to check that, because if you are iterating over all the possible git extensions, since there are some of them that are not supported by jgit, then you will get always git as recommended tool

Reply:
That is true, I assume that we only get the user-chosen extension list. I believe the GitSCM constructor does so, anyways I need to interactively test this feature, I've been taking some time to create the extensions right now, will do that and then confirm.

Reply:
let me know your results!

Reply:
@fcojfernandez I was able to interactively test this addition on a local instance. I created a sample Freestyle project with git scm to checkout a < 5 MiB repo. 

The GitToolChooser suggested "JGit" for this project. 
After adding additional behavior to the project configuration: 

- Case 1: With `Additional Checkout options` -> timeout = 10 mins
Since timeout is not supported by "JGit", GitToolChooser suggested "git" for the project. 

- Case 2: With `Additional Clone Behavior` -> Enable Shallow Clone
Not supported feature, "git" was recommended.

Reply:
good to know that the extension lists is the real list of extensions for the job